src/Lykke.Messaging.RabbitMq/RabbitMqTransportFactory.cs

```diff
@@ -84,31 +84,22 @@ public ITransport Create(ILog log, TransportInfo transportInfo, Action onFailure
 
         public ITransport Create(TransportInfo transportInfo, Action onFailure)
         {
-            if (_transports.TryGetValue(transportInfo, out var result))
-                return result;
-
-            lock (_sync)
+            return _transports.GetOrAdd(transportInfo, ti =>
             {
-                if (_transports.TryGetValue(transportInfo, out result))
-                    return result;
-
-                var brokers = transportInfo.Broker
+                var brokers = ti.Broker
                     .Split(',')
                     .Select(b => b.Trim())
                     .ToArray();
 
-                result = new RabbitMqTransport(
+                return new RabbitMqTransport(
```

Comment:
Is ```new RabbitMqTransport()``` heavy operation?

Reply:
No, it's not heavy

Reply:
Ok, but note that multiple ```RabbitMqTransport``` instances can be created for the same ```transportInfo```

Reply:
That's ok